analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -45,97 +52,71 @@ public Package(string id, string version, string runtime)
                 }
-                else
+
+                // Check to see if the specific package is already installed
+                var packageDir = PackageDirectory();
+                if (!Directory.Exists(packageDir))
                 {
-                    var packageDir = PackageDirectory();
-                    if (Directory.Exists(packageDir))
-                    {
-                        LogMessage($"Package found at {packageDir}");
-                    }
-                    else
+                    LogMessage($"Package not found at {packageDir}, will attempt to download and install.");
+                    InstallPackageAsync(writeTimeStamp).Wait();
+                    if (!Directory.Exists(packageDir))
                     {
-                        LogMessage($"Package not found at {packageDir}");
-                        Install();
+                        throw new ApplicationException($"Test setup error: folder for downloaded package does not exist. Folder: {packageDir}");
                     }
                 }
             }
 
-            private void Install()
+            private async Task InstallPackageAsync(bool writeTimeStamp)
             {
-                var versionArgument = Version == Constants.NuGetLatestVersion ? string.Empty : $"-Version {Version}";
-                var configFile = ValidatedNuGetConfigPath();
-                // Explicitly specify the NuGet config to use to avoid being impacted by the NuGet config on the machine running the tests.
-                // If the unit tests are failing since the nuget packages cannot be installed (nuget.exe: ERROR: Illegal characters in path.)
-                // please ensure that the PackagesFolder does not end with a separator.
-                var args = $"install {Id} {versionArgument} -OutputDirectory \"{Path.GetFullPath(PackagesFolder)}\" -NonInteractive -ForceEnglishOutput -ConfigFile \"{configFile}\"";
-                LogMessage($"Installing package using nuget.exe {args}");
-                var startInfo = new ProcessStartInfo
-                {
-                    FileName = "nuget.exe",
-                    Arguments = args,
-                    UseShellExecute = false,
-                    CreateNoWindow = true,
-                    RedirectStandardError = true,
-                    RedirectStandardOutput = true
-                };
-
-                using (var process = new Process())
-                {
-                    process.StartInfo = startInfo;
-                    process.OutputDataReceived += (s, e) => LogMessage($"  nuget.exe: {e.Data}");
-                    process.ErrorDataReceived += OnErrorDataReceived;
+                var resource = await GetNuGetRepository();
 
-                    process.Start();
-                    process.BeginErrorReadLine();
-                    process.BeginOutputReadLine();
+                using var packageStream = new MemoryStream();
+                await resource.CopyNupkgToStreamAsync(
```

Comment:
Is this too long to be a single line?
`CancellationToken.None` can be shortened to `default`